content/developer/reference/guidelines.rst

```diff
@@ -999,17 +999,88 @@ More precise JS guidelines are detailed in the `github wiki  <https://github.com
 You may also have a look at existing API in Javascript by looking Javascript
 References.
 
-CSS coding guidelines
----------------------
+Template guidelines
+-------------------
+
+Structure
+~~~~~~~~~
+
+.. note:: These rules are in place because the general structure of the template is considered more important than the details of each line.
+
+- Each block element should be on its own line.
+- Small inline/style/text elements can be placed either on the same line as their parent or on their own line (depending on their importance and size).
+- Indentation of 4 spaces is applied between a parent and its children elements. The closing tag of a parent should always be at the same indentation level as its opening tag.
+- It is preferable to use instructions (such as `t-if`) directly on the element on which they apply (as opposed to having them on a separate `<t>` element).
+
+Naming
+~~~~~~
+
+The root element and each other *important* element of the template should have a `name` attribute to ease the readability of the template.
```

Comment:
The problem with using comments is that they are not targetable in tests or xpath (not that I like doing them, but it helps the community too), and also they are not present in the resulting DOM for easier debugging.

The problem with using class for naming, if we use bootstrap or similar approach to style, is that class attribute becomes extremely noisy, and the name is for us immensely more important to understand the template than many style values.
Therefore a new separate attribute is proposed.

But as you noted it is optional so I don't think it matters too much.

Reply:
Are you actually suggesting to use a random "name" attribute on random elements I am not sure? :thinking: If that is the case, that is not a valid attribute for any HTML element so this should definitely not be done... if I understood well :grimacing:

Reply:
I wanted another syntax completely but AL wanted name, so this was a compromise. There are already various `name=` in templates all around Odoo code and I don't think it makes any issue?
Technically it's a valid attribute, but only on form elements.

We can use `data-name` or anything else, but it will be more verbose.
Another suggestion would be `t-name`, or another built-in instruction, in that case we can print in the DOM something different than what we have in the template, so we can make the syntax less verbose in the template (no need for the component part) while still having the expected result in DOM (but AL/FP don't like template not strictly equal DOM).

Reply:
a comment is also valid html, and is formatted in a different color, which makes it easier to see. It is also totally standard and anyone outside of odoo will know what it is.  And as @qsm-odoo said, `name` is not a valid html attribute. Also, if your template is difficult to understand, I am not sure how an additional attribute will help that much.

Reply:
> Technically it's a valid attribute, but only on form elements. [...] and I don't think it makes any issue

So it's not a valid attribute on any HTML element. Although we have some errors to fix since a while, passing W3C validation is still something we want on the website side. Probably less important in the backend but still louche to go against W3C rules for no reason :wink: